pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -569,6 +569,12 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1beta1.PipelineRun, get
 	}
 
 	if pipelineRunFacts.State.IsBeforeFirstTaskRun() {
+		if err := resources.ValidateMatrixPipelineParameterTypes(pipelineRunFacts.State); err != nil {
+			logger.Errorf("Failed to validate matrix %q with error %v", pr.Name, err)
+			pr.Status.MarkFailed(ReasonInvalidTaskResultReference, err.Error())
```

Comment:
the failure reason is `ReasonInvalidTaskResultReference`, but that's not relevant here -- need to find an existing relevant reason or add a new one